internal/builder/builder_test.go

```diff
@@ -237,6 +245,38 @@ func testWithoutTestSourceNotInGoPath(t *testing.T) {
 	builderTC.run(t)
 }
 
+func testWithIncorrectProjectStructure(t *testing.T) {
+	dir := filepath.Join(os.TempDir(), t.Name(), "godogs")
+	files := map[string]string{
+		"godogs.go": emptyBuilderTestFile,
+		"go.mod":    builderModFile,
+	}
+
+	err := buildTestPackage(dir, files)
+	defer os.RemoveAll(dir)
+	require.Nil(t, err)
+
+	prevDir, err := os.Getwd()
+	require.Nil(t, err)
+	err = os.Chdir(dir)
+	require.Nil(t, err)
+	defer os.Chdir(prevDir)
+
+	testBin, err := filepath.Abs(filepath.Join(dir, "godog.test"))
+	require.Nil(t, err)
+
+	if build.Default.GOOS == "windows" {
+		testBin += ".exe"
+	}
+
+	// call the builder - we should get an error
+	err = builder.Build(testBin)
+	// check that we even got an error at all
+	require.NotNil(t, err)
+	// now check the details of the error message
+	require.Contains(t, err.Error(), "incorrect project structure")
```

Comment:
```suggestion
	require.EqualError(t, err, "incorrect project structure: no test files found")
```
Let's make error assertion more detailed.